data_creation/data_creation.py

```diff
@@ -0,0 +1,357 @@
+            print(f"relative_start_frame: {relative_start_frame}, relative_end_frame: {relative_end_frame}")
+
+            # Ensure relative frame range is within the sentence's local frame range
+            if relative_start_frame < 0 or relative_end_frame > len(pose_body.data):
+                print(f"Skipping Gloss '{gloss}' due to frame range out of bounds.")
+                continue
+
+            valid_gloss_count += 1
+
+
+            # Extract original pose for current gloss
+            original_pose_body = pose_body.select_frames(range(relative_start_frame, relative_end_frame))
+            original_pose_header = self._get_pose_header("dgs_corpus")
+            original_pose = Pose(original_pose_header, original_pose_body)
+            original_poses_sequence.append(original_pose)
+
+
+            # Determine if the gloss should be replaced
+            if self._should_replace(gloss):
+                print(f"Original pose shape: {original_pose_body.data.shape}")
+                replaced_gloss_count += 1
+                types_gloss_pose_path = self.dictionary[gloss]['views']['pose']
+                with open(types_gloss_pose_path, "rb") as f:
+                    types_gloss_pose = Pose.read(f.read())
+
+                    # original_frame_count = len(original_pose_body.data)
+                    # current_frame_count = len(types_gloss_pose.body.data)
+                    # print(f"Original frame count: {original_frame_count}")
+                    # print(f"Current frame count: {current_frame_count}")
+                    print(f"Updated pose shape before interpolation: {types_gloss_pose.body.data.shape}")
+
+                    # Interpolate pose if fps of dgs types pose is different from dgs corpus pose
+                    if types_gloss_pose.body.fps != original_pose_body.fps:
+                        interpolated_pose_body = types_gloss_pose.body.interpolate(new_fps=original_pose_body.fps)
+                        updated_pose_header = self._get_pose_header("dgs_types")
+                        updated_pose = Pose(updated_pose_header, interpolated_pose_body)
+                        print(f"Updated pose shape after interpolation: {interpolated_pose_body.data.shape}")
+                    else:
+                        updated_pose = types_gloss_pose
+            else:
+                updated_pose = original_pose
+
+            updated_poses_sequence.append(updated_pose)
+
+
+        # Concatenate poses
+        concatenated_original_pose = concatenate_poses(original_poses_sequence)
+        concatenated_updated_pose = concatenate_poses(updated_poses_sequence)
+
+        # print(f"Concatenated original pose shape: {concatenated_original_pose.body.data.shape}")
+        # print(f"Concatenated updated pose shape: {concatenated_updated_pose.body.data.shape}")
+
+        return concatenated_original_pose, concatenated_updated_pose, metadata, valid_gloss_count, replaced_gloss_count
```

Comment:
returning a naked tuple, makes it so you later access the data in an unreadable way: `processed_data[3]` (unclear what is 3?)

Instead, return a typed object, such as:
```py
from dataclasses import dataclass

@dataclass
class Point:
    x: int
    y: int = 1  # Set default value

Point(3)  # -> Point(x=3, y=1)
```